docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,293 @@
+Input attributes are used to match the rule entry with the SKR request and to modify [search labels](#search-labels). Output attributes are only used to modify the [search labels](#search-labels).
+In its minimal form each rule consists of a `PLAN` that the rule applies to.
+In its extended form a rule entry contains lists of input attributes and its values passed in the form of `ATTR=VAL` pairs in parantheses.
+
+> ![NOTE]
+> The following rule entries are considered valid and the same:
+> * `aws`
+> * `aws()`
+
+Output attributes does not support values.
+Description of when the rule is triggered can be found in the [Rule Evaluation](#evaluation) section.
+Rule Attributes are described in [Rule Attributes](#rule-attributes) section. 
+
+Currently, possible OUTPUT_ATTR_x attributes values are `S` or `EU`.
+
+```
+hap: 
+  rule: 
+  - azure(INPUT_ATTR_1=VAL_1,INPUT_ATTR_2=VAL_2,...,INPUT_ATTR_N) -> OUTPUT_ATTR_1=VAL_1,OUTPUT_ATTR_2=VAL_2,...,OUTPUT_ATTR_N
+```
+
+The input attribute types include **platformRegion** (`PR`), **hyperscalerRegion** (`HR`). The output attribute types include **shared** (`S`) and **euAccess** (`EU`). 
+Each attribute can be used at most once in a single rule entry.
+**shared** and **euAccess** attributes are used only to apply their labels modifications if rest of matched attributes are equal to SKR's values. 
+
+Rule Entry results include `S` and `EU`. They are described in [Rule Results](#rule-results) section.
+
+## Rule Evaluation
+
+HAP evaluates a set of rules to determine what labels to use when querying secret bindings.
+Rule is evaluated during cluster provisioning.
+Rule entries are analyzed one by one.
+Physical attributes from every entry are compared with values from input, if they are the same the rule entry is considered triggered.
+If there is more than one triggered rule then only one is selected in accordance to priority rules described in the [Priority](#priority) section.
+If there are no rule entries trigered then the error is returned. No fallback behaviour is defined in such case.
+
+## Uniqnuess & Priority
+
+Only one rule can be triggered.
+
+If more than one rule entry match the request than only one is filterd out and applied. 
+
+The process of selecting the best matching rule is based on a rule uniqnues and priority.
+
+Rule entry uniqnuess is determined by its plan and input parameters with its values (identification attributes) not including input parameters with `*`. 
+
+Output parameters are not taken into account when establishing rule entry uniqnuess. For example the following rule will fail on startup because every skr can be matched by both rules:
+
+```
+hap:
+  rule: 
+    - gcp 
+    - gcp -> S # invalid entry, output attributes does not take part into uniqnuess check
+    - gcp(PR=*) # invalid entry, both can be applied to the same skr
+```
+
+The similary behaviour occurs when the rule defined the same attributes values together with `*` value.
```

Comment:
Description is not clear however the example is. I will try to rephrase.

Reply:
similar
defines

Reply:
If I am not mistaken the point is that the same SKR could match two rules, which is prohibited.
So this is the key, not using `*` per se.